test/sys/test_socket.rs

```diff
@@ -1527,6 +1543,60 @@ pub fn test_unixdomain() {
     assert_eq!(&buf[..], b"hello");
 }
 
+// Test using unnamed unix domain addresses
+#[cfg(any(target_os = "android", target_os = "linux"))]
+#[test]
+pub fn test_unnamed_unixdomain() {
+    use nix::sys::socket::{getsockname, socketpair};
+    use nix::sys::socket::{SockFlag, SockType};
+    use nix::unistd::close;
+
+    let (fd_1, fd_2) = socketpair(
+        AddressFamily::Unix,
+        SockType::Stream,
+        None,
+        SockFlag::empty(),
+    )
+    .expect("socketpair failed");
+
+    let addr_1: UnixAddr = getsockname(fd_1).expect("getsockname failed");
+    assert!(addr_1.is_unnamed());
+    assert_eq!(addr_1, UnixAddr::new_unnamed());
```

Comment:
Is this part guaranteed to work, according to any Linux documentation, or is it just luck?  I'm not sure we should have this assertion.

Reply:
Linux says "When the address of an unnamed socket is returned, its length is `sizeof(sa_family_t)`", so the only possible unnamed unix socket address it could return would be `sockaddr_un { sun_family: AF_UNIX, ... }` with length 2. But I agree that the documentation around this still isn't great, so I removed this assertion.